applications/GeoMechanicsApplication/custom_retention/retention_law.h

```diff
@@ -84,20 +82,19 @@ class KRATOS_API(GEO_MECHANICS_APPLICATION) RetentionLaw {
             return mFluidPressure.value();
         }
 
-        const ProcessInfo& GetProcessInfo() const
+        [[nodiscard]] static double CalculateFluidPressure(const Vector& rN, const Vector& rPressureVector)
```

Comment:
I fully agree with the idea to move this functionality to a single location, rather than it being scattered across several elements (as it used to be). I doubt, however, whether this nested class is the best possible location. Perhaps we should consult @WPK4FEM. I can imagine that class `GeoElementUtilities` might be a good alternative. What do you think gentlemen?

Reply:
I think we discussed it briefly. Otherwise I would place the function in transport_..._utilities. GeoElementUtilities includes a lot of various functions right now. @WPK4FEM , what is your opinion?

Reply:
Computing an interpolated value of a nodal variable uses the interpolation polynomials. These belong to the element and not to constitutive laws nor to retention laws. Therefore GeoElementUtilities sounds like a good place. utilities for retention laws or constitutive models is a bad place. Transport Equation utilities is also a good place, because it handles element matrices for the water pressure part.

As interpolation by definition is the inner product or shape functions and nodal variables, just writing inner_prod is fine. Naming this operation specifically for the water pressures is better, because the name of the function is explaining what the intention is.

Reply:
Thank you for the detailed answer. The function has been moved to transport_equation_utilities.